### PR TITLE
Entertaining Issue #41 (Request) Mass evolving settings (Pidgey Hoard…

### PR DESCRIPTION
**Pull request by SiBangkotan**
This issue was marked unnecesary already, but after the ban wave, a lot has changed. I see the code at this repository had nicely added separate setting for Evolve (Animation) Delay. This delay create a new behavior. Unlike the previous insta-evolve, now with default value supplied by "global.settings" at 30 second animation delay, then you got max 60 evolve for each lucky egg session. That's why they need this fixed number parameter. I've seen someone else asking for this same feature at other unmaintained repository branch. With random catch of pokemons, they find it difficult to guess a percentage that matched lucky egg time and evolve animation delay.

I just put half a line code in the evolve logic, so they start evolving when storage percentage reached, or when this number reached, whichever occur first. The rest of the change is just adjusting interfaces for the parameter to appear in config.json.

It's good to see that the config.json is very neat as compared to the previous one.

### CODE REVIEW

File: PoGo.NecroBot.Logic/Interfaces/Configuration/ILogicSettings.cs [hunk 1]
```diff
@@ -133,6 +133,7 @@ public interface ILogicSettings
         bool VerboseRecycling { get; }
         double RecycleInventoryAtUsagePercentage { get; }
         double EvolveKeptPokemonsAtStorageUsagePercentage { get; }
+        int OrWhicheverFirst_EvolveKeptPokemonsAtThisNumber { get; }
```
Comment (samuraitruong) [inline PoGo.NecroBot.Logic/Interfaces/Configuration/ILogicSettings.cs]: IT not a nice naming convention, please change the name more readable

**Comment by SiBangkotan** [comment]
Thanks for reminding me that parameter name is out of rhyme. Done changed.

Changed default value as well to 120 at multiple of two lucky eggs at default Evolve Animation Delay 30 seconds. This rather high value of default is meant to let the "Storage Percentage" still dictate the bot behavior if end user not changing this number. Other possible value is 180, or even 250 that will practically disable this trigger completely unless end user change the value.

Upper bound check in global.setting increased a bit to allow "pokemon storage item" thing, if any.

**Comment by jjskuld** [comment]
This seems like a pretty clean change.  I think since there official client allows something like this then we should have no problems with this change. In fact Nia changed the evolution time down to around 18 seconds from what I just read so you'll be able to evolve even more now. Since people also shutdown their official app and relaunch it to go even faster than 28 seconds per evolve. I think with newer phones that can relaunch official app within 4-5 seconds.

**Comment by jjskuld** [comment]
Close in favor of #413

**Pull request closed without merging**